### ISSUE TITLE
Update Translations with transifex

### ISSUE THREAD

**Issue by Eselce** [greasemonkey/greasemonkey#2853]
I've updated one moved and 7 new german strings in transifex.com.
Thanks for uploading the english original (not too late).

Some remarks:
- git doesn't like last lines without newline, thus I've added `\n` at the end
- What about the placeholders? There are none in the commit, can transifex handle them?
- Thus I get the wrong output:
```
Benutzerskript nicht ausführbar: [Greasemonkey script ; version ]
Error: executeScript requires either a 'code' or a 'file' property, but not both
```
or (with the correct strings from transifex.com):
```
Benutzerskript nicht ausführbar: [Greasemonkey Benutzerskript ; Version ]
Error: executeScript requires either a 'code' or a 'file' property, but not both
```
The placeholders are substituted with empty strings.

When I add the placeholder part to the entry:
```
    "[Greasemonkey script $id$; version $version$]": {
        "message": "[Greasemonkey Benutzerskript $ID$; Version $VERSION$]",
        "description": "",
        "placeholders": {
            "id": {
                "content": "$1",
                "example": "Name des Benutzerskripts"
            },
            "version": {
                "content": "$2",
                "example": "1.0"
            }
        }
    },
```
it works just fine:
```
Benutzerskript nicht ausführbar: [Greasemonkey Benutzerskript null/Unbenanntes Skript 500975; Version 1]
Error: executeScript requires either a 'code' or a 'file' property, but not both
```

**Comment by arantius** [comment]
I tried re-writing the pull to do better/easier translation and I got at least one detail wrong.  I think there's some upper/lower case place holder issues?

**Comment by Eselce** [comment]
Doesn't matter, it works this was (the key was `$id$` and `$version$` while the text was `$ID$` and `$VERSION$` and the placeholders `id` and `version`. I've tested that, it works (but maybe the same case would be better).

**Comment by Eselce** [comment]
The only placeholder issue right now is, that the downloaded file doesn't contain any placeholders, and even more not translations for them. The placeholders in the strings are substituted by empty strings!

**Comment by Sxderp** [comment]
Perhaps do without named placeholders? I can understand for longer strings or have more substitutions but this one seems fairly simple.

**Comment by Eselce** [comment]
These placeholders were requested by @arantius - and there are even some more examples left. Time to solve the general problem...

**Comment by arantius** [comment]
I'm interested in feedback about how to do this well.  I'm trying to balance making both the code itself make sense with the translators' work (usually in isolation of the code) also making sense.

**Comment by Eselce** [comment]
I don't know all the answers, that's why I asked. I'd also like to hear the opinions of other translators.
Foremost, I'd like to have a proper workflow, communication channel and safe review.

Here are some more remarks:
- The 8th new string "New user script..." could easily be derived from former "New user script ..." (stands for de, ja and zh_CN) without consulting the translators.
- Unused or future translations are superflous, but they don't crash anything.
- I would appreciate having code independent translations, but as usual, this is hard: I needed to look at the code and even a running system to get the german stuff right. For example, if you have "Runs on:", how should you know, that this is the label for @include patterns?
- There's a way to do this properly: Once for all, someone has to fill the "description" part of the messages (but I guess, transifex would ignore it)
- I don't see, how this can be done unmoderated...

**Comment by arantius** [comment]
AFAICT everything here merged; please continue discussion in #2862.
